docs/proposals/20220725-managed-kubernetes.md

```diff
@@ -0,0 +1,493 @@
+last-updated: 2022-07-25
+status: implementable
+see-also:
+replaces:
+superseded-by:
+---
+
+# Managed Kubernetes in CAPI
+
+## Table of Contents
+
+- [Managed Kubernetes in CAPI](#managed-kubernetes-in-capi)
+  - [Table of Contents](#table-of-contents)
+  - [Glossary](#glossary)
+  - [Summary](#summary)
+  - [Motivation](#motivation)
+    - [Goals](#goals)
+    - [Non-Goals/Future Work](#non-goalsfuture-work)
+  - [Proposal](#proposal)
+    - [User Stories](#user-stories)
+      - [Story 1](#story-1)
+      - [Story 2](#story-2)
+      - [Story 3](#story-3)
+      - [Story 4](#story-4)
+      - [Story 5](#story-5)
+      - [Story 6](#story-6)
+    - [Current State of Managed Kubernetes in CAPI](#current-state-of-managed-kubernetes-in-capi)
+      - [EKS in CAPA](#eks-in-capa)
+      - [AKS in CAPZ](#aks-in-capz)
+    - [Managed Kubernetes API Design Approaches](#managed-kubernetes-api-design-approaches)
+      - [Option 1: Single kind for Control Plane and Infrastructure](#option-1-single-kind-for-control-plane-and-infrastructure)
+        - [Background: Why did EKS in CAPA choose this option?](#background-why-did-eks-in-capa-choose-this-option)
+      - [Option 2: Two kinds with a ControlPlane and a pass-through InfraCluster](#option-2-two-kinds-with-a-controlplane-and-a-pass-through-infracluster)
+      - [Option 3: Two kinds with a Managed Control Plane and Managed Infra Cluster with Better Separation of Responsibilities](#option-3-two-kinds-with-a-managed-control-plane-and-managed-infra-cluster-with-better-separation-of-responsibilities)
+      - [Option 4: Two kinds with a Managed Control Plane and Shared Infra Cluster with Better Separation of Responsibilities](#option-4-two-kinds-with-a-managed-control-plane-and-shared-infra-cluster-with-better-separation-of-responsibilities)
+  - [Recommendations](#recommendations)
+    - [Additional notes on option 3](#additional-notes-on-option-3)
+    - [Managed Node Groups for Worker Nodes](#managed-node-groups-for-worker-nodes)
+    - [Provider Implementers Documentation](#provider-implementers-documentation)
+  - [Other Considerations for CAPI](#other-considerations-for-capi)
+    - [ClusterClass support for MachinePool](#clusterclass-support-for-machinepool)
+    - [clusterctl integration](#clusterctl-integration)
+    - [Add-ons management](#add-ons-management)
+  - [Upgrade Strategy](#upgrade-strategy)
+  - [Implementation History](#implementation-history)
+  
+## Glossary
+
+- **Managed Kubernetes** - a Kubernetes service offered/hosted by a service provider where the control plane is run & managed by the service provider. As a customer you don’t have to worry about managing/operating the control plane machines. Additionally, the managed Kubernetes service may extend to cover running managed worker nodes. Examples are EKS in AWS and AKS in Azure. This is different from a traditional Kubernetes implementation, where the control plane and worker nodes are deployed and managed by the customer.
```

Comment:
Thanks, this glossary is great. Just one comment, instead of using "As a customer" or "where the user is" where customer and user are ambiguous terms can we agree on specific personas which are enabled by Managed Kubernetes? e.g.

Cluster Service Provider
The user hosting cluster control planes, responsible for up-time. UI for fleet wide alerts, configuring AWS account to host control planes in, views user provisioned infra (available compute). Has cluster admin management.

Cluster Service Consumer
The user empowered to request control planes, request workers, and drive upgrades or modify externalized configuration.

Cluster Instance Admin
The user with cluster-admin role in the provisioned cluster, but may have no power over when/how cluster is upgraded or configured.

Cluster (Instance) User
Maps to a developer.

Reply:
Great suggestion @enxebre :+1:  I like this.

Reply:
Great suggestion. I will read the proposal again and make changes for the personas.

Reply:
Added personas and updated user story as best as I can 😀